gwtquery-core/src/main/java/com/google/gwt/query/client/GQuery.java

```diff
@@ -278,24 +225,51 @@ public TagWrapper(int wrapDepth, String preWrap, String postWrap) {
       if (o instanceof String) {
         return $((String) o);
       }
+      if (o instanceof SafeHtml) {
+        return $(((SafeHtml) o).asString());
+      }
       if (o instanceof GQuery) {
         return (GQuery) o;
       }
       if (o instanceof Function) {
-        return $((Function) o);
-      }
-      if (JsUtils.isElement(o)) {
-        return $(JsUtils.<Element> cast(o));
+        return new GQuery(((Function) o).getElement());
       }
       if (o instanceof JsonBuilder) {
         return new GQuery(((JsonBuilder) o).<Element>getDataImpl());
       }
-      if (o instanceof JavaScriptObject) {
-        return $((JavaScriptObject) o);
-      }
       if (o instanceof IsWidget) {
         return $(Arrays.asList(o));
       }
+      if (o instanceof JavaScriptObject) {
+        JavaScriptObject jso = (JavaScriptObject) o;
+        // Execute a native javascript function like jquery does
+        if (JsUtils.isFunction(jso)) {
+          new JsUtils.JsFunction(jso).fe();
+          return $();
+        }
+        // Wraps a native array like jquery does
+        if (!JsUtils.isWindow(jso) && !JsUtils.isElement(jso) && JsUtils.isArray(jso)) {
+          JsArrayMixed c = jso.cast();
+          JsNodeArray elms = JsNodeArray.create();
+          for (int i = 0; i < c.length(); i++) {
+            Object obj = c.getObject(i);
+            if (obj instanceof Node) {
+              elms.addNode((Node) obj);
+            }
+          }
+          return new GQuery(elms);
+        }
+        // Wraps a native NodeList
+        if (JsUtils.isNodeList(jso)) {
+          return new GQuery(jso.<NodeList<Element>> cast());
+        }
+        // Wraps an event
+        if (JsUtils.isEvent(jso)) {
+          jso = jso.<Event>cast().getCurrentEventTarget();
+        }
+        // Otherwise we wrap it as an element
+        return new GQuery(jso.<Element> cast());
+      }
       console
```

Comment:
It seems the problem is not because the jdt, but the GWT compiler implementation of J8. Since it is not complete yet, it does not behaves exactly equal than java compiler, and although the code looks ok in IDE and compiles in JVM, overloaded evaluation is more restrictive in GWT.